### ISSUE TITLE
Train vs Test dataset reconstructions for Autoencoder

### ISSUE THREAD

**Issue by michaelklachko** [phlippe/uvadlc_notebooks#58]
In the [Tutorial 9](https://uvadlc-notebooks.readthedocs.io/en/latest/tutorial_notebooks/tutorial9/AE_CIFAR10.html) when comparing latent dimensionality, the plots show reconstruction results on the train dataset. If the model overfits (for example, if someone decides to make the model bigger) train image reconstruction quality might become misleading. I recommend using test dataset images for that experiment.

**Comment by phlippe** [comment]
Hi, as far as I see in Tutorial 9, the plot for the reconstruction loss shows the validation performance (`model_dict[k]["result"]["val"][0]["test_loss"]`), not the reconstruction on the training dataset. Did you mean the difference between validation and test set? Since we use this plot to choose the 'best' latent size for later application, this comparison should be rather understood as a hyperparameter search and hence no test dataset should be ideally involved. Nonetheless, the difference between the two sets is anyway negligible since the autoencoders are very small in comparison to the dataset size.

**Comment by michaelklachko** [comment]
I meant the code in cell 13:

```
input_imgs = get_train_images(4)
for latent_dim in model_dict:
    visualize_reconstructions(model_dict[latent_dim]["model"], input_imgs)
```

I agree that for hyperparameter search it's best to use validation set, rather than test set, but in this case it appears the images from are picked from the train set. Sure, it's not important for this model, but if I make it 10x bigger, overfitting does happen, and the reconstruction quality between train and test images starts to differ.

**Comment by phlippe** [comment]
Ah, ok. Yes, with larger models and if it is a crucial decision, the validation set should be used there for the qualitative results, and many more images since we just aim at an intuition here with 4 images. I will add a comment in the notebooks to clarify this. I prefer keeping the original images since they align with all other images I used in CIFAR10 throughout this repository.
